internal/uvm/network.go

```diff
@@ -96,6 +108,29 @@ func (uvm *UtilityVM) AddNetNS(ctx context.Context, id string) error {
 	return nil
 }
 
+// AddEndpointToNS adds an ednpoint to the network namespace with the specified
+// NIC ID.
+//
+// If no network namespace matches `id` returns `ErrNetNSNotFound`.
+func (uvm *UtilityVM) AddEndpointToNS(ctx context.Context, nsID, nicID string, endpoint *hns.HNSEndpoint) error {
```

Comment:
can we rename this? I don't like that we have one function called `AddEndpointToNS` and another with only an 's' added.

Reply:
I can just use AddEndpointsToNS also. I just didn't like constructing a single elem slice for no reason. What would you recommend this be named if it stays.

Reply:
In that case, I would prefer we only use one function or the other.

Reply:
I'll change :)

Reply:
I'm going to keep this actually. I'm realizing another reason this was added after revisiting, we don't pass in a Nic ID for `AddEndpointsToNS` we always just generate it. How does `AddEndpointToNSWithID` sound?